### PR TITLE
Fix dismissed observable in UIViewController

### PR DESCRIPTION
**Pull request by dmanuelcl**
## Description
Sometime the UIViewController is removed from parent without call dismiss(animated:completion:) for example when  the UIViewController is managed by UINavigationController and call UINavigationController.setViewControllers(_:animated:)

## Checklist
- [x] this PR is based on develop or a 'develop related' branch
- [x] the commits inside this PR have explicit commit messages
- [ ] the Jazzy documentation has been generated (if needed -> Jazzy RxFlow)

### CODE REVIEW

File: RxFlow/Extensions/UIViewController+Rx.swift [hunk 1]
```diff
@@ -28,7 +28,11 @@ extension Reactive where Base: UIViewController {
             .filter { _ in self.base.isBeingDismissed }
             .map { _ in false }
 
-        return ControlEvent(events: source)
+        let source2 = self.sentMessage(#selector(Base.didMove(toParentViewController:)))
```
Comment (twittemb) [inline RxFlow/Extensions/UIViewController+Rx.swift]: Could you rename source and source2 in something more meaningful please ? perhaps dismissedSource and movedToParentSource ?

**Comment by twittemb** [comment]
Your modifications brings some memory leaks in the demo application. 

Could you rework that please ? thanks a lot 👍

capture d ecran le 2018-07-15 a 21 13 05

File: RxFlow/Extensions/UIViewController+Rx.swift (hunk 1 above)
Comment (dmanuelcl) [reply RxFlow/Extensions/UIViewController+Rx.swift]: Ok, no problem, I'm going to do the necessary tests and rename source and source2

Comment (twittemb) [reply RxFlow/Extensions/UIViewController+Rx.swift]: Great thanks a lot.

**Comment by dmanuelcl** [comment]
Sorry for the delay I've been a bit busy, I've done several tests and I do not think memory problems are the cause of my changes, if you follow the flow that I will attach you will see that without my changes you get the same problems, it should be noted that these problems only occur in 40% of the tests that I can perform with and without my changes

The problems point me to

```swift
   self.identifier = "\(type(of: flow))-\(Date())"
```
when you initialize FlowCoordinator
 and

```swift
static var sceneStoryboard: UIStoryboard {
     return UIStoryboard (name: String (describing: self), bundle: Bundle (for: self))
   }
```
in the Reusable pod

I really do not think there's any problem with those code snippets

rxflowmalloc

Excuse my English

**Comment by twittemb** [comment]
thanks @dmanuelcl 

Indeed the memory issues seem a bit random and not related to your modifications. For sure something I will track in the future.

See ya 👍.

**Comment by dmanuelcl** [comment]
Thanks to you for your excellent work
👍

**Pull request merged**